x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/MultilingualE5SmallInternalServiceSettings.java

```diff
@@ -8,42 +8,107 @@
+import org.elasticsearch.xpack.inference.services.ServiceFields;
 import org.elasticsearch.xpack.inference.services.ServiceUtils;
 import org.elasticsearch.xpack.inference.services.settings.InternalServiceSettings;
 
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.Map;
+import java.util.Objects;
 
+import static org.elasticsearch.TransportVersions.ML_INFERENCE_L2_NORM_SIMILARITY_ADDED;
 import static org.elasticsearch.TransportVersions.ML_TEXT_EMBEDDING_INFERENCE_SERVICE_ADDED;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractSimilarity;
 
 public class MultilingualE5SmallInternalServiceSettings extends ElasticsearchInternalServiceSettings {
 
     public static final String NAME = "multilingual_e5_small_service_settings";
 
-    public MultilingualE5SmallInternalServiceSettings(int numAllocations, int numThreads, String modelId) {
+    static final int DIMENSIONS = 384;
+    static final SimilarityMeasure DEFAULT_SIMILARITY = SimilarityMeasure.COSINE;
+
+    private final SimilarityMeasure similarity;
+    private final int dimensions;
+
+    public MultilingualE5SmallInternalServiceSettings(
+        int numAllocations,
+        int numThreads,
+        String modelId,
+        SimilarityMeasure similarityMeasure,
+        int dimensions
+    ) {
         super(numAllocations, numThreads, modelId);
+
+        this.similarity = similarityMeasure;
+        this.dimensions = dimensions;
     }
 
     public MultilingualE5SmallInternalServiceSettings(StreamInput in) throws IOException {
         super(in.readVInt(), in.readVInt(), in.readString());
+
+        if (in.getTransportVersion().onOrAfter(TransportVersions.ML_INFERENCE_L2_NORM_SIMILARITY_ADDED)) {
+            similarity = in.readEnum(SimilarityMeasure.class);
+            dimensions = in.readVInt();
+        } else {
+            similarity = DEFAULT_SIMILARITY;
+            dimensions = DIMENSIONS;
+        }
     }
 
     /**
-     * Parse the MultilingualE5SmallServiceSettings from map and validate the setting values.
+     * Parse the MultilingualE5SmallServiceSettings from map that originated from persistent storage and validate the setting values.
      *
      * If required setting are missing or the values are invalid an
      * {@link ValidationException} is thrown.
      *
      * @param map Source map containing the config
      * @return The {@code MultilingualE5SmallServiceSettings} builder
      */
-    public static MultilingualE5SmallInternalServiceSettings.Builder fromMap(Map<String, Object> map) {
+    public static MultilingualE5SmallInternalServiceSettings.Builder fromPersistentMap(Map<String, Object> map) {
         ValidationException validationException = new ValidationException();
+        var requestFields = extractRequestFields(map, validationException);
+
+        Integer parsedDimensions = ServiceUtils.removeAsType(map, ServiceFields.DIMENSIONS, Integer.class);
+        if (parsedDimensions == null) {
```

Comment:
When the inference endpoint is created we will not allow the user to specify the dimensions. This code is for parsing persisted configuration.